src/PotentialSimulation/SimulationAlgorithms/SOR.jl

```diff
@@ -24,3 +24,28 @@ end
     end 
     nothing
 end
+
+
+function _guess_optimal_number_of_threads_for_SOR(gs::NTuple{3, Integer}, max_nthreads::Integer, S::Union{Type{Cylindrical}, Type{Cartesian}})::Int
+    max_nthreads = max_nthreads > Base.Threads.nthreads() ? Base.Threads.nthreads() : max_nthreads
+    n = S == Cylindrical ? gs[2] * gs[3] : gs[1] * gs[2] # Number of grid points to be updated in each iteration of the outer loop
```

Comment:
Not sure whether this is more performant than the `?`-`:`- syntax but just as an inspiration:
```julia
n = gs[2] * g[1 + 2 * (S==Cylindrical)]
```

Reply:
I also don't whether this is more performant. However, this is not a high performant critically code anyway. 
I would keep the current implementation due to better (clearer) readability.